src/main.rs

```diff
@@ -114,6 +166,45 @@ async fn update_receipt_status(
             }
         }
     }
+    info!(target: INDEXER_FOR_WALLET, "update_receipt_status finished");
+}
+
+async fn handle_outcomes(outcomes: &Vec<near_indexer::Outcome>, pool: &Pool<ConnectionManager<PgConnection>>) {
+    let mut failed_receipt_ids: Vec<String> = vec![];
+    let mut succeeded_receipt_ids: Vec<String> = vec![];
+
+    for outcome in outcomes {
+        match outcome {
+            near_indexer::Outcome::Receipt(execution_outcome) => {
+                match execution_outcome.outcome.status {
+                    near_indexer::near_primitives::views::ExecutionStatusView::Failure(_) => failed_receipt_ids.push(execution_outcome.id.to_string()),
+                    near_indexer::near_primitives::views::ExecutionStatusView::SuccessReceiptId(_)
+                    | near_indexer::near_primitives::views::ExecutionStatusView::SuccessValue(_) => {
+                        succeeded_receipt_ids.push(execution_outcome.id.to_string());
+                    },
+                    _ => {}
+                }
+            },
+            _ => {},
+        }
+    }
+
+    info!(
+        target: INDEXER_FOR_WALLET,
+        "Failed outcomes, amount {}", failed_receipt_ids.len()
+    );
+
+    let update_failed_future = update_receipt_status(failed_receipt_ids, ExecutionStatus::Failed, &pool);
+
+
+    info!(
+        target: INDEXER_FOR_WALLET,
+        "Succeed outcomes, amount {}", succeeded_receipt_ids.len()
+    );
```

Comment:
Given we don't execute the futures immediately, it makes sense to merge the two log messages into a single one.